src/server/hset_family.cc

```diff
@@ -1005,8 +1006,37 @@ void HSetFamily::HStrLen(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void StrVecEmplaceBack(StringVec& str_vec, const listpackEntry& lp) {
+  if (lp.sval) {
+    str_vec.emplace_back(reinterpret_cast<char*>(lp.sval), lp.slen);
+  } else {
+    str_vec.emplace_back(absl::StrCat(lp.lval));
+  }
+  return;
+}
+
 void HSetFamily::HRandField(CmdArgList args, ConnectionContext* cntx) {
+  if (args.size() > 3) {
+    DVLOG(1) << "Wrong number of command arguments: " << args.size();
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+
   string_view key = ArgS(args, 0);
+  int32_t count;
+  string_view with_values;
+
+  if (args.size() > 1) {
+    if (!SimpleAtoi(ArgS(args, 1), &count)) {
+      return (*cntx)->SendError("count value is not an integer", kSyntaxErrType);
+    }
+  }
+
+  if (args.size() == 3) {
+    ToUpper(&args[2]);
+    with_values = ArgS(args, 2);
+    if (with_values != "WITHVALUES")
+      return (*cntx)->SendError(kSyntaxErr);
+  }
```

Comment:
That is a _very_ unusual way to set a boolean flag

just do `if (ArgS(args, 2) != "W...S") return error; with_values = true;`

Reply:
`with_values` here is not a boolean flag, but i see what you mean. Will change it to a boolean value.